index.js

```diff
@@ -1,29 +1,74 @@
-/*NOTE: the size of the files extracted from the <input type="file"> element
-            is measured in Bytes. Thus, all convertions from Bytes, to KiloBytes, MegaBytes 
-            and so on must take that into account. */
-
 const KB = 1024;
 const MB = KB*KB;
 const GB = KB*MB;
 
+const supported_extensions = ["jpg",
+    "jpeg","png","gif","bmp","tiff",
+    "svg","webp","ico","heif","avif",
+    "JPEG","PNG","GIF","BMP","TIFF",
+    "SVG","WEBP","ICO","HEIF","AVIF",
+    "JPG"];
+
 const AVAIL_SPACE = "available_space";
 const size_suffix = "&&fileSize";
 
-const total_space = 10*MB;/*measured in Byte units.*/
+const total_space = 10*MB;
+let available_space;
+
+const size_to_string = size_bytes => {
+    if(size_bytes/GB>1){
+        return `${(size_bytes/GB).toFixed(1)} GB`;
+    }
+    if(size_bytes/MB>1){
+        return `${(size_bytes/MB).toFixed(1)} MB`;
+    }
+    if(size_bytes/KB>1){
+        return `${(size_bytes/KB).toFixed(1)} KB`;
+    }
+    return `${size_bytes} Bytes`;
+}
+
+const sizeKey = fileName => `${fileName}${size_suffix}`;
+
+const setAvailableSpace = num => localStorage.setItem(AVAIL_SPACE,num);
+
+const setBarWidth = () => {
+    let percentage = 100*(total_space-available_space)/total_space;
+    document.getElementsByClassName("gradient-bar")[0].setAttribute("style",`width:${percentage}%`);
+}
+
+const updateUsed = () =>document.getElementById("used-space").innerText = size_to_string(total_space - available_space);
+
+const updateRemaining = () => {
+    const num_and_unit = size_to_string(available_space).split(" ");
+    document.getElementById("remaining-space").innerHTML = 
+    `${num_and_unit[0]} <span>${num_and_unit[1]} left</span>`;
+}
+
+const updateHTML =() => {
+    setBarWidth();
+    updateRemaining();
+    updateUsed();
+}
+
+
+const removeFile = fileName => {
+    localStorage.removeItem(fileName);
+    let size = localStorage.getItem(sizeKey(fileName))
+    available_space += parseInt(size);
```

Comment:
You could extract the logic of your local storage into separate functions and call them: 

```js
removeFileFromStorage(fileName);
getFileSize(fileName)
```